sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -530,7 +527,7 @@ private class BulkWriter(container: CosmosAsyncContainer,
           SMono.empty
       })
 
-      if (Exceptions.isTimeout(statusCode)) {
+      if (Exceptions.isTimeout(statusCode) && requestTimeoutDelayInjected.compareAndSet(false, true)) {
```

Comment:
hmm, if we only inject the timeout for the first operation, then would it defeat the purpose that for high CPU caused timeout, we want to slow down the traffic?

for example, we get 100 response with all timeout, only one of them will be added delay, and the other 99 operations will be retried right away, is this the expected behavior here?

Reply:
Yes, tiemout would never ever happen individually for multiple operations - because tiemout in backend is 5 seconds - at whihc point the timeout would happen for all operations

Reply:
So, worst case batch request with 100 operations - 20 succeed - then 21 times out and 21-99 report 408 or some other error

Reply:
without ordering guarantee you could have some succeeding and some failing of the 100 - but bottom-line the CPU and scope of timing out is always the batch - not the individual operation

Reply:
Fixed - removed the 408 retry changes - they aren't strictly necessary and probably good to go over them separately.